lib/src/git_backend.rs

```diff
@@ -1818,6 +1876,50 @@ mod tests {
         assert_eq!(std::str::from_utf8(&sig.data).unwrap(), commit_str);
     }
 
+    #[test]
+    fn round_trip_change_id_via_git_header() {
```

Comment:
nit: Is the round-trip actually via the header? If I'm reading the code correctly, we update `commit.change_id` in `deserialize_extras()` after we fill it in based on the commit header. So I think the round-trip is actually via the proto storage.

I think we should already make that assignment conditional on the proto having a (non-empty) value for the change id. That should simplify a future transition where we stop writing to the proto storage. Otherwise we'll run into the same thing as we did with the `jj:trees` header in #6186. I.e. otherwise some old jj binaries (but built after this PR) will expect overwrite the change id by an empty change id when newer binaries stop writing to the proto storage.

Reply:
You are right that the proto shadows the change-id being loaded in this test. Besides not loading empty proto values, I've also made so that we stop populating the change id in the proto storage too.

Reply:
That does make sense. How about preventing loading it from the proto if the setting is enabled instead? There isn't a great way to test this otherwise because the proto load shadows it.

Reply:
> How about preventing loading it from the proto if the setting is enabled instead?

That would work in the test but not for existing repos out there, right?

> There isn't a great way to test this otherwise because the proto load shadows it.

Maybe the test can re-init the jj repo using the existing Git repo? If the change-id header is in the Git repo, it should be preserved. Would that work?

Reply:
That's what the git fetch tests did that we removed earlier basically. I can implement that here instead though.

Reply:
much simpler to do it here, just a empty store and a populated git backend